File: backend/app/schemas/character.py
from pydantic import BaseModel
from typing import Optional, Any, Dict

class CharacterBase(BaseModel):
    name: str
    role: Optional[str] = None
    description: Optional[str] = None
    visual_tags: Optional[Dict[str, Any]] = None
    avatar_url: Optional[str] = None
    turnaround_url: Optional[str] = None
    face_url: Optional[str] = None
    model_type: Optional[str] = "pony"  # "pony" or "flux"

class CharacterCreate(CharacterBase):
    project_id: int

class CharacterUpdate(CharacterBase):
    name: Optional[str] = None
    project_id: Optional[int] = None

class Character(CharacterBase):
    id: int
    project_id: int

    class Config:
        from_attributes = True

File: backend/src/schemas/character.ts
import { z } from 'zod';

export const CharacterSchema = z.object({
  id: z.number().int(),
  project_id: z.number().int(),
  name: z.string(),
  role: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
  visual_tags: z.union([z.string(), z.record(z.string(), z.any())]).nullable().optional(),
  avatar_url: z.string().nullable().optional(),
  turnaround_url: z.string().nullable().optional(),
  face_url: z.string().nullable().optional(),
  model_type: z.string().default('pony')
});

export const CharacterCreateSchema = z.object({
  project_id: z.number().int(),
  name: z.string(),
  role: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
  visual_tags: z.union([z.string(), z.record(z.string(), z.any())]).nullable().optional(),
  avatar_url: z.string().nullable().optional(),
  turnaround_url: z.string().nullable().optional(),
  face_url: z.string().nullable().optional(),
  model_type: z.string().default('pony')
});

export const CharacterUpdateSchema = z.object({
  project_id: z.number().int().optional(),
  name: z.string().optional(),
  role: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
  visual_tags: z.union([z.string(), z.record(z.string(), z.any())]).nullable().optional(),
  avatar_url: z.string().nullable().optional(),
  turnaround_url: z.string().nullable().optional(),
  face_url: z.string().nullable().optional(),
  model_type: z.string().optional()
});

export type Character = z.infer<typeof CharacterSchema>;
export type CharacterCreate = z.infer<typeof CharacterCreateSchema>;
export type CharacterUpdate = z.infer<typeof CharacterUpdateSchema>;
